File: luogu_code/introduction_5_struct/P5735.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

typedef struct point{
    double x, y;
}point;

double cal_distance(point p1, point p2){
    return sqrt((p1.x - p2.x) * (p1.x - p2.x) + (p1.y - p2.y) * (p1.y - p2.y));
}

double cal_circumference(point p1, point p2, point p3){
    return cal_distance(p1, p2) + cal_distance(p2, p3) + cal_distance(p3, p1);
}

int main(){
    point p1, p2, p3;
    cin >> p1.x >> p1.y >> p2.x >> p2.y >> p3.x >> p3.y;
    cout << fixed << setprecision(2) << cal_circumference(p1, p2, p3) << endl;
    return 0;
}

File: luogu_code/introduction_6_struct/P5735/P5735.py
class Point:
    def __init__(self,x,y):
        self.x = x
        self.y = y
        
def cal_distance(p1,p2):
    return ((p1.x - p2.x) ** 2 + (p1.y - p2.y) ** 2) ** 0.5

def cal_circumference(p1,p2,p3):
    a = cal_distance(p1,p2)
    b = cal_distance(p2,p3)
    c = cal_distance(p3,p1)
    return a + b + c

def main():
    x1,y1 = map(float,input().split())
    x2,y2 = map(float,input().split())
    x3,y3 = map(float,input().split())
    p1 = Point(x1,y1)
    p2 = Point(x2,y2)
    p3 = Point(x3,y3)
    print(f"{cal_circumference(p1,p2,p3):.2f}")

if __name__ == "__main__":
    main()
